### ISSUE TITLE
Bad error message when downloading private model without being logged in.

### ISSUE THREAD

**Issue by patrickvonplaten** [huggingface/transformers#15980]
Let's say an organization creates a private model and wants to share it with other team members which are less savy of `huggingface_hub` and `transformers`. 

So e.g. I create: https://huggingface.co/NewT5/dummy_model

and want to share it with others.

Now if I run:

```python
from transformers import BertModel

BertModel.from_pretrained("NewT5/dummy_model")
```

I'm getting a very nice error message:

```
OSError: NewT5/dummy_model is not a local folder and is not a valid model identifier listed on 'https://huggingface.co/models'
If this is a private repository, make sure to pass a token having permission to this repo with `use_auth_token` or log in with `huggingface-cli login` and pass `use_auth_token=True`.
```

After this error message I think people will have an easy time doing the correct thing which is passing **use_auth_token=True** and previously running `huggingface-cli login`.

Now what will often happen though in my opinion is that someone will share the following code with unsavy coworkers / collaborators:

```python
from transformers import BertModel

BertModel.from_pretrained("NewT5/dummy_model", use_auth_token=True)
```

Now **if you are not logged in**, you are getting the following error message:

```
OSError: Can't load config for 'NewT5/dummy_model'. If you were trying to load it from 'https://huggingface.co/models', make sure you don't have a local directory with the same name. Otherwise, make sure 'NewT5/dummy_model' is the correct path to a directory containing a config.json file
```

This error message is not great really, because the problem is not that the model doesn't exist, but it's because the user didn't run `huggingface-cli login`

I think it's worth fixing the error message here (maybe just the same as when passing `use_auth_token=True` is missing because IMO it's a common case that people will hare code with `use_auth_token=True`.

We probably need to do this in moon-landing though no? 

## Env

- `transformers` version: 4.18.0.dev0
- Platform: Linux-5.15.15-76051515-generic-x86_64-with-glibc2.34
- Python version: 3.9.7
- PyTorch version (GPU?): 1.10.0 (False)
- Tensorflow version (GPU?): 2.8.0 (False)
- Flax version (CPU?/GPU?/TPU?): 0.3.6 (cpu)
- Jax version: 0.2.25
- JaxLib version: 0.1.73

and hugging face hub version:

`0.4.0.dev0`

**Comment by patrickvonplaten** [comment]
Related: https://github.com/huggingface/datasets/issues/3855

**Comment by SBrandeis** [comment]
> We probably need to do this in moon-landing though no?

I don't think that requires changes to moon-landing

From what i understand, you need to catch [this error](https://github.com/huggingface/transformers/blob/master/src/transformers/file_utils.py#L2117) and display the appropriate message [here](https://github.com/huggingface/transformers/blob/master/src/transformers/configuration_utils.py#L629) (currently the underlying error message is suppressed)

**Comment by sgugger** [comment]
To catch [this error](https://github.com/huggingface/transformers/blob/master/src/transformers/file_utils.py#L2117), the `huggingface_hub` would need to issue a proper subclass of `EnvironmentError` (for instance `HFNotLoggedInError`).

Of course defaulting `use_auth_token` to `True` would be easier, but that's something that has been debated several times already.

**Comment by LysandreJik** [comment]
In my mind `huggingface_hub` already had support for these errors, will fix the issue upstream in `huggingface_hub`.

**Comment by LysandreJik** [comment]
Will be solved by https://github.com/huggingface/huggingface_hub/pull/878

**Comment by tejasri19** [comment]
You can see what models are available by going to [Models - Hugging Face 1.0k](https://huggingface.co/models) and using the filter by name:

**Comment by hgeorgescu** [comment]
How can I get a login at the private repository detaining TFBertForSequenceClassification?
